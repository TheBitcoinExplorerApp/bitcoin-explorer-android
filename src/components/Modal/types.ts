import { BlockType } from "../BlocksInfo/components/Block/type";
import { inputTransactions, outputTransactions } from "../Transactions/types";

type BasicModalProps = {
  modalType: "Transaction" | "Block" | "Address" | "Information";
  isVisible: boolean;
  handleModalClose: () => void;
};

export type InformationModalProps = BasicModalProps & {
  modalType: "Information";
};

export type BlockModalProps = BasicModalProps & {
  modalType: "Block";
  blockTransactions: TransactionType[];
} & BlockType;

export type TransactionModalProps = BasicModalProps & {
  modalType: "Transaction";
  transactionInfo: TransactionType;
  transactionHash: string;
};

export type TransactionType = {
  transactionId: string;
  size: number;
  fee: number;
  inputTransactions: inputTransactions[];
  outputTransactions: outputTransactions[];
  statusTransaction: {
    confirmed: boolean;
    blockHeight?: number;
    blockHash?: string;
    blockTime?: number;
  };
};

export type AddressInfoType = {
  address: string;
  totalAmountReceived: number;
  totalAmountSent: number;
  balance: number;
};

export type AddressModalProps = BasicModalProps & {
  modalType: "Address";
  addressInfo: {
    address: string;
    addressData: AddressInfoType;
  };
  addressTransactions: TransactionType[];
};

export type ModalProps =
  | BlockModalProps
  | TransactionModalProps
  | AddressModalProps;
