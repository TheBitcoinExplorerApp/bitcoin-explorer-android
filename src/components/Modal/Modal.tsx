/* eslint-disable react/jsx-props-no-spreading */ // #TODO: VERIFY IF IS POSSIBLE TO NOT BE SPREAD PROPS
import { ModalProps } from "./types";
import TransactionModal from "./components/TransactionModal/TransactionModal";
import BlockModal from "./components/BlockModal/BlockModal";
import AddressModal from "./components/AddressModal/AddressModal";

export default function Modal(props: ModalProps) {
  const { modalType } = props;

  if (modalType === "Transaction") {
    return <TransactionModal {...props} />;
  }

  if (modalType === "Address") {
    return <AddressModal {...props} />;
  }

  return <BlockModal {...props} />;
}
