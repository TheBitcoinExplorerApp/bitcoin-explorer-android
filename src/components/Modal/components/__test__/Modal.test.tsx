/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import {
  initialStateAddress,
  initialStateBlocks,
  initialStateTransaction,
} from "src/__mocks__/initialStates";
import Modal from "../../Modal";

describe("Modal", () => {
  it("renders correctly when type is Transaction", () => {
    const tree = renderer
      .create(
        <Modal
          isVisible
          modalType="Transaction"
          transactionHash="#ABC12"
          handleModalClose={() => {}}
          transactionInfo={initialStateTransaction}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when type is Address", () => {
    const tree = renderer
      .create(
        <Modal
          isVisible
          modalType="Address"
          addressInfo={{
            address: "0x123",
            addressData: initialStateAddress,
          }}
          handleModalClose={() => {}}
          addressTransactions={[
            initialStateTransaction,
            initialStateTransaction,
          ]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when type is Block", () => {
    const tree = renderer
      .create(
        <Modal
          isVisible
          modalType="Block"
          handleModalClose={() => {}}
          size={initialStateBlocks[0].size}
          extras={initialStateBlocks[0].extras}
          timeAgo={initialStateBlocks[0].timeAgo}
          blockHash={initialStateBlocks[0].blockHash}
          blockHeight={initialStateBlocks[0].blockHeight}
          satPerVbyte={initialStateBlocks[0].satPerVbyte}
          transactions={initialStateBlocks[0].transactions}
          blockTransactions={[initialStateTransaction, initialStateTransaction]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
