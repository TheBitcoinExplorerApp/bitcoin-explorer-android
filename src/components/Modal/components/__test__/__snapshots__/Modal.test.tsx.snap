// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly when type is Address 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  style={
    {
      "flexDirection": "row",
    }
  }
  transparent={true}
  visible={true}
>
  <RCTScrollView
    contentContainerStyle={
      {
        "height": "auto",
        "marginTop": "38%",
        "paddingBottom": "40%",
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#101427",
            "height": "100%",
            "paddingHorizontal": 10,
            "paddingVertical": 17,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
              }
            }
          >
            Endereço
          </Text>
          <View
            accessibilityLabel="Close Modal"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#D9D9D9",
                "borderRadius": 100,
                "height": 36,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 5,
                "position": "absolute",
                "right": 0,
                "width": 36,
              }
            }
          >
            <svg
              height={20}
              width={20}
            />
          </View>
        </View>
        <View
          style={
            {
              "gap": 20,
              "height": "100%",
              "marginTop": 35,
              "paddingHorizontal": 8,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1d2133",
                    "height": 48,
                    "justifyContent": "space-between",
                    "paddingHorizontal": 18,
                    "paddingVertical": 9,
                  },
                  {
                    "borderBottomColor": undefined,
                    "borderBottomEndRadius": undefined,
                    "borderBottomStartRadius": undefined,
                    "borderBottomWidth": 0,
                    "borderRadius": 7,
                    "borderTopEndRadius": undefined,
                    "borderTopStartRadius": undefined,
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#FFF",
                    "fontSize": 15,
                  }
                }
              >
                Endereço
              </Text>
              <Text
                style={
                  {
                    "color": "#DF7800",
                    "fontSize": 15,
                  }
                }
              >
                0x123...
              </Text>
            </View>
          </View>
          <View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1d2133",
                    "height": 48,
                    "justifyContent": "space-between",
                    "paddingHorizontal": 18,
                    "paddingVertical": 9,
                  },
                  {
                    "borderBottomColor": undefined,
                    "borderBottomEndRadius": 0,
                    "borderBottomStartRadius": 0,
                    "borderBottomWidth": 0,
                    "borderRadius": 7,
                    "borderTopEndRadius": undefined,
                    "borderTopStartRadius": undefined,
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#FFF",
                    "fontSize": 15,
                  }
                }
              >
                Total Recebido
              </Text>
              <Text
                style={
                  {
                    "color": "#DF7800",
                    "fontSize": 15,
                  }
                }
              >
                0 BTC
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1d2133",
                    "height": 48,
                    "justifyContent": "space-between",
                    "paddingHorizontal": 18,
                    "paddingVertical": 9,
                  },
                  {
                    "borderBottomColor": undefined,
                    "borderBottomEndRadius": 0,
                    "borderBottomStartRadius": 0,
                    "borderBottomWidth": 0,
                    "borderRadius": 7,
                    "borderTopEndRadius": 0,
                    "borderTopStartRadius": 0,
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#FFF",
                    "fontSize": 15,
                  }
                }
              >
                Total Enviado
              </Text>
              <Text
                style={
                  {
                    "color": "#DF7800",
                    "fontSize": 15,
                  }
                }
              >
                0 BTC
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1d2133",
                    "height": 48,
                    "justifyContent": "space-between",
                    "paddingHorizontal": 18,
                    "paddingVertical": 9,
                  },
                  {
                    "borderBottomColor": undefined,
                    "borderBottomEndRadius": undefined,
                    "borderBottomStartRadius": undefined,
                    "borderBottomWidth": 0,
                    "borderRadius": 7,
                    "borderTopEndRadius": 0,
                    "borderTopStartRadius": 0,
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#FFF",
                    "fontSize": 15,
                  }
                }
              >
                Saldo
              </Text>
              <Text
                style={
                  {
                    "color": "#DF7800",
                    "fontSize": 15,
                  }
                }
              >
                0 BTC
              </Text>
            </View>
          </View>
          <View>
            <View
              style={
                {
                  "backgroundColor": "#101427",
                  "gap": 16,
                  "height": "100%",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#FFF",
                    "fontSize": 15,
                    "fontWeight": "600",
                  }
                }
              >
                Entradas e Saídas
              </Text>
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1d2133",
                    "borderRadius": 8,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "maxWidth": "100%",
                    "paddingVertical": 5,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "gap": 5,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#DF7800",
                          "fontSize": 16,
                        }
                      }
                    >
                       ...
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#FFF",
                          "fontSize": 13,
                          "textAlign": "right",
                        }
                      }
                    >
                      0
                       BTC
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    {
                      "color": "#D9D9D9",
                      "fontSize": 28,
                      "fontWeight": "800",
                    }
                  }
                >
                  &gt;
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "gap": 5,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#DF7800",
                          "fontSize": 16,
                        }
                      }
                    >
                       ...
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#FFF",
                          "fontSize": 13,
                          "textAlign": "right",
                        }
                      }
                    >
                      0
                       BTC
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "backgroundColor": "#101427",
                  "gap": 16,
                  "height": "100%",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#FFF",
                    "fontSize": 15,
                    "fontWeight": "600",
                  }
                }
              >
                Entradas e Saídas
              </Text>
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1d2133",
                    "borderRadius": 8,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "maxWidth": "100%",
                    "paddingVertical": 5,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "gap": 5,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#DF7800",
                          "fontSize": 16,
                        }
                      }
                    >
                       ...
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#FFF",
                          "fontSize": 13,
                          "textAlign": "right",
                        }
                      }
                    >
                      0
                       BTC
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    {
                      "color": "#D9D9D9",
                      "fontSize": 28,
                      "fontWeight": "800",
                    }
                  }
                >
                  &gt;
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "gap": 5,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#DF7800",
                          "fontSize": 16,
                        }
                      }
                    >
                       ...
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#FFF",
                          "fontSize": 13,
                          "textAlign": "right",
                        }
                      }
                    >
                      0
                       BTC
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</Modal>
`;

exports[`Modal renders correctly when type is Block 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RCTScrollView
    contentContainerStyle={
      {
        "backgroundColor": "#101427",
        "height": "100%",
        "marginTop": "38%",
        "paddingBottom": "40%",
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#101427",
            "height": "100%",
            "paddingHorizontal": 10,
            "paddingVertical": 17,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
              }
            }
          >
            Bloco
          </Text>
          <View
            accessibilityLabel="Close Modal"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#D9D9D9",
                "borderRadius": 100,
                "height": 36,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 5,
                "position": "absolute",
                "right": 0,
                "width": 36,
              }
            }
          >
            <svg
              height={20}
              width={20}
            />
          </View>
        </View>
        <View
          style={
            {
              "gap": 20,
              "height": "100%",
              "marginTop": 35,
              "paddingHorizontal": 8,
            }
          }
        >
          <ActivityIndicator
            color="#FFF"
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</Modal>
`;

exports[`Modal renders correctly when type is Transaction 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  style={
    {
      "flexDirection": "row",
    }
  }
  transparent={true}
  visible={true}
>
  <RCTScrollView
    contentContainerStyle={
      {
        "backgroundColor": "#101427",
        "height": "auto",
        "marginTop": "38%",
        "paddingBottom": "40%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#101427",
            "paddingHorizontal": 10,
            "paddingVertical": 17,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
              }
            }
          >
            Transação
          </Text>
          <View
            accessibilityLabel="Close Modal"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#D9D9D9",
                "borderRadius": 100,
                "height": 36,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 5,
                "position": "absolute",
                "right": 0,
                "width": 36,
              }
            }
          >
            <svg
              height={20}
              width={20}
            />
          </View>
        </View>
        <View
          style={
            {
              "gap": 16,
              "marginTop": 35,
              "paddingHorizontal": 8,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1d2133",
                    "height": 48,
                    "justifyContent": "space-between",
                    "paddingHorizontal": 18,
                    "paddingVertical": 9,
                  },
                  {
                    "borderBottomColor": undefined,
                    "borderBottomEndRadius": undefined,
                    "borderBottomStartRadius": undefined,
                    "borderBottomWidth": 0,
                    "borderRadius": 7,
                    "borderTopEndRadius": undefined,
                    "borderTopStartRadius": undefined,
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#FFF",
                    "fontSize": 15,
                  }
                }
              >
                Hash
              </Text>
              <Text
                style={
                  {
                    "color": "#DF7800",
                    "fontSize": 15,
                  }
                }
              >
                #ABC12...
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "gap": 16,
              }
            }
          >
            <View
              style={
                {
                  "gap": 16,
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#1d2133",
                        "height": 48,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 18,
                        "paddingVertical": 9,
                      },
                      {
                        "borderBottomColor": undefined,
                        "borderBottomEndRadius": undefined,
                        "borderBottomStartRadius": undefined,
                        "borderBottomWidth": 0,
                        "borderRadius": 7,
                        "borderTopEndRadius": undefined,
                        "borderTopStartRadius": undefined,
                        "flexDirection": "row",
                        "width": "auto",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#FFF",
                        "fontSize": 15,
                      }
                    }
                  >
                    Não Confirmada
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#DF7800",
                        "fontSize": 15,
                      }
                    }
                  />
                </View>
              </View>
              <View>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#1d2133",
                        "height": 48,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 18,
                        "paddingVertical": 9,
                      },
                      {
                        "borderBottomColor": "#D9D9D9",
                        "borderBottomEndRadius": 0,
                        "borderBottomStartRadius": 0,
                        "borderBottomWidth": 0.1,
                        "borderRadius": 7,
                        "borderTopEndRadius": undefined,
                        "borderTopStartRadius": undefined,
                        "flexDirection": "row",
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#FFF",
                        "fontSize": 15,
                      }
                    }
                  >
                    Data/Hora
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#DF7800",
                        "fontSize": 15,
                      }
                    }
                  >
                    Aguardando Confirmação
                  </Text>
                </View>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#1d2133",
                        "height": 48,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 18,
                        "paddingVertical": 9,
                      },
                      {
                        "borderBottomColor": "#D9D9D9",
                        "borderBottomEndRadius": 0,
                        "borderBottomStartRadius": 0,
                        "borderBottomWidth": 0.1,
                        "borderRadius": 7,
                        "borderTopEndRadius": 0,
                        "borderTopStartRadius": 0,
                        "flexDirection": "row",
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#FFF",
                        "fontSize": 15,
                      }
                    }
                  >
                    Tamanho
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#DF7800",
                        "fontSize": 15,
                      }
                    }
                  >
                    0 B
                  </Text>
                </View>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#1d2133",
                        "height": 48,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 18,
                        "paddingVertical": 9,
                      },
                      {
                        "borderBottomColor": undefined,
                        "borderBottomEndRadius": undefined,
                        "borderBottomStartRadius": undefined,
                        "borderBottomWidth": 0,
                        "borderRadius": 7,
                        "borderTopEndRadius": 0,
                        "borderTopStartRadius": 0,
                        "flexDirection": "row",
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#FFF",
                        "fontSize": 15,
                      }
                    }
                  >
                    Taxa
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#DF7800",
                        "fontSize": 15,
                      }
                    }
                  >
                    0 BTC
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "backgroundColor": "#101427",
                  "gap": 16,
                  "height": "100%",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#FFF",
                    "fontSize": 15,
                    "fontWeight": "600",
                  }
                }
              >
                Entradas e Saídas
              </Text>
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1d2133",
                    "borderRadius": 8,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "maxWidth": "100%",
                    "paddingVertical": 5,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "gap": 5,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#DF7800",
                          "fontSize": 16,
                        }
                      }
                    >
                       ...
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#FFF",
                          "fontSize": 13,
                          "textAlign": "right",
                        }
                      }
                    >
                      0
                       BTC
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    {
                      "color": "#D9D9D9",
                      "fontSize": 28,
                      "fontWeight": "800",
                    }
                  }
                >
                  &gt;
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "gap": 5,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#DF7800",
                          "fontSize": 16,
                        }
                      }
                    >
                       ...
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#FFF",
                          "fontSize": 13,
                          "textAlign": "right",
                        }
                      }
                    >
                      0
                       BTC
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</Modal>
`;
